impl/src/planning/subplanners.rs

```diff
@@ -636,23 +636,77 @@ impl<'planner> CrateSubplanner<'planner> {
   }
 
   /// Generates source details for internal crate.
-  fn produce_source_details(&self, package: &Package, package_root: &Path) -> SourceDetails {
-    SourceDetails {
-      git_data: self.source_id.as_ref().filter(|id| id.is_git()).map(|id| {
+  fn produce_source_details(
+    &self,
+    crates_io_template: &str,
+    package: &Package,
+    package_root: &Path,
+  ) -> Result<SourceDetails> {
```

Comment:
If this now returns a `Result`, should we cleanup the use of `unwrap` here?

Reply:
Good call - done